pkg/servicequota/eipusage.go

```diff
@@ -0,0 +1,71 @@
+package servicequota
+
+import (
+	"fmt"
+
+	"github.com/oslokommune/okctl/pkg/config"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/servicequotas"
+	"github.com/logrusorgru/aurora"
+	"github.com/oslokommune/okctl/pkg/api/okctl.io/v1alpha1"
+)
+
+type eipusage struct {
+	CloudProvider v1alpha1.CloudProvider
+}
+
+func (e eipusage) Count() (int, error) {
+	eips, err := e.CloudProvider.EC2().DescribeAddresses(nil)
+	if err != nil {
+		return -1, fmt.Errorf("failed to get eip count: %w", err)
+	}
+
+	return getLengthOf(getStringMapOf(eips.String()), "Addresses")
+}
+
+func (e eipusage) Quota() (int, error) {
+	quotas, err := e.CloudProvider.ServiceQuotas().GetServiceQuota(&servicequotas.GetServiceQuotaInput{
+		QuotaCode:   aws.String("L-0263D0A3"),
+		ServiceCode: aws.String("ec2"),
+	})
+	if err != nil {
+		return -1, fmt.Errorf("failed to get eip quota: %w", err)
+	}
+
+	return getQuotaNumericValue(quotas)
+}
+
+func (eipusage) Required() int {
+	return config.DefaultRequiredEpis
```

Comment:
We should probably inject the required number of "things" when newing up the struct.